src/Chainweb/Miner.hs

```diff
@@ -125,6 +141,37 @@ toMinerData = MinerData . encodeToByteString
 {-# INLINABLE toMinerData  #-}
 
 -- | Convert from Chainweb 'MinerData' to Pact Miner
+--
 fromMinerData :: MonadThrow m => MinerData -> m Miner
 fromMinerData = decodeStrictOrThrow' . _minerData
 {-# INLINABLE fromMinerData #-}
+
+-- | Rewards table mapping 3-month periods to their rewards
+-- according to the calculated exponential decay over 120 year period
+--
+readRewards
+    :: HasChainGraph v
+    => v -> HashMap BlockHeight ParsedDecimal
+readRewards v =
+    case CSV.decode CSV.NoHeader (toS rawMinerRewards) of
+      Left e -> error
+        $ "cannot construct miner reward map: "
+        <> sshow e
+      Right vs -> HM.fromList
+        . V.toList
+        . V.map formatRow
+        $ vs
+  where
+    formatRow :: (Word64, Double) -> (BlockHeight, ParsedDecimal)
+    formatRow (!a,!b) =
+      let
+        !n = v ^. chainGraph . to (int . size)
+        !m = fromRational $ toRational b
+      in (BlockHeight a, ParsedDecimal $ m / n)
+{-# INLINE readRewards #-}
```

Comment:
Is this used more than once during the lifetime of a node? Since this seems somewhat expensive, maybe we shouldn't inline it but memoize it instead?

If it's called only once, it's fine. Still, it maybe better to not inline it.

Reply:
It's only called once, so it should be okay